pkcs11/CMakeLists.txt

```diff
@@ -78,12 +78,21 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set_target_properties(yubihsm_pkcs11 PROPERTIES LINK_FLAGS "-liconv")
 endif()
 
-target_link_libraries(
-  yubihsm_pkcs11
-  yubihsm
-  ${LIBCRYPTO_LDFLAGS}
-  ${GETOPT_LIBS}
-  )
+if(ENABLE_STATIC)
+  target_link_libraries(
+    yubihsm_pkcs11
+    yubihsm_static
+    ${LIBCRYPTO_LDFLAGS}
+    ${GETOPT_LIBS}
+    )
+else(ENABLE_STATIC)
+  target_link_libraries(
+    yubihsm_pkcs11
+    yubihsm
+    ${LIBCRYPTO_LDFLAGS}
+    ${GETOPT_LIBS}
+    )
+endif(ENABLE_STATIC)
```

Comment:
If you wish, I'm quite sure you can have multiple additions, e.g.
```
target_link_libraries(x a b c)
if (ENABLE_STATIC)
  target_link_libraries(x d_static)
else()
  target_link_libraries(x d)
endif()
```

AFAIK the argument of `else()` and `endif()` can be (nowadays) omitted.

I don't know much about this project: are some of these dependencies eligible as `PRIVATE`?

Reply:
Thanks for the review, I have updated as you suggested. Not sure what you mean by PRIVATE here